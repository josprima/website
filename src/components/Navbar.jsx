import PropTypes from 'prop-types';
import React, { useState, useEffect } from 'react';
import classnames from 'classnames';
import Link from 'next/link';

import Logo from '../assets/logo.svg';
import css from './Navbar.module.scss';

const isMobileView = () => {
  const viewPortWidth = Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);

  return viewPortWidth < 600;
};

const LinkItem = ({ text, url }) => (
  <li className={css.navbar__item}>
    <Link href={url}>
      <a className={css.navbar__link}>{text}</a>
    </Link>
  </li>
);

const Navbar = () => {
  const [isOpen, setIsOpen] = useState(false);

  const handleResize = () => {
    if (!isMobileView() && isOpen) {
      setIsOpen(false);
    }
  };

  useEffect(() => {
    window.addEventListener('resize', handleResize);

    return () => window.removeEventListener('resize', handleResize);
  }, [isMobileView]);

  const buttonClass = classnames(
    css.navbar__toggle,
    {
      [css['navbar__toggle--close']]: isOpen,
    },
  );

  const sideMenuClass = classnames(
    css['navbar__side-menu'],
    {
      [css['navbar__side-menu--open']]: isOpen,
    },
  );

  return (
    <section className={css.navbar}>
      <Logo />

      <ul className={css.navbar__menu}>
        <LinkItem text="Home" url="/" />
        <LinkItem text="Alur Belajar" url="/" />
        <LinkItem text="Akun" url="/" />
      </ul>

      <button
        className={buttonClass}
        type="button"
        onClick={() => setIsOpen(!isOpen)}
        data-testid="button"
      >
        <div />
        <div />
        <div />
      </button>

      <ul className={sideMenuClass}>
        <LinkItem text="Home" url="/" />
        <LinkItem text="Alur Belajar" url="/" />
        <LinkItem text="Akun" url="/" />
      </ul>
    </section>
  );
};

LinkItem.propTypes = {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default Navbar;
