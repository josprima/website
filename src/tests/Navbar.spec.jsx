import { render, fireEvent } from '@testing-library/react';
import Navbar from '../components/Navbar';

test('renders a message', () => {
  const { container, getByTestId } = render(<Navbar />);
  expect(container).toMatchSnapshot();

  fireEvent.click(getByTestId('button'));
});
