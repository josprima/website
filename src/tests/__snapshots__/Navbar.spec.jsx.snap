// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a message 1`] = `
<div>
  <section
    class="navbar"
  >
    <div />
    <ul
      class="navbar__menu"
    >
      <li
        class="navbar__item"
      >
        <a
          class="navbar__link"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="navbar__item"
      >
        <a
          class="navbar__link"
          href="/"
        >
          Alur Belajar
        </a>
      </li>
      <li
        class="navbar__item"
      >
        <a
          class="navbar__link"
          href="/"
        >
          Akun
        </a>
      </li>
    </ul>
    <button
      class="navbar__toggle"
      data-testid="button"
      type="button"
    >
      <div />
      <div />
      <div />
    </button>
    <ul
      class="navbar__side-menu"
    >
      <li
        class="navbar__item"
      >
        <a
          class="navbar__link"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="navbar__item"
      >
        <a
          class="navbar__link"
          href="/"
        >
          Alur Belajar
        </a>
      </li>
      <li
        class="navbar__item"
      >
        <a
          class="navbar__link"
          href="/"
        >
          Akun
        </a>
      </li>
    </ul>
  </section>
</div>
`;
